### PR TITLE
PEP 670: Rearrange and reword the PEP

### PR DESCRIPTION
**Pull request by encukou**
@vstinner & @erlend-aasland, this should make the PEP more understandable & complete

### CODE REVIEW

File: pep-0670.rst
Comment (vstinner) [inline pep-0670.rst:85]: Just add Py_PYTHON_H as another example on the first item "Empty macros. Example: #define Py_HAVE_CONDVAR.", no?

File: pep-0670.rst
Comment (vstinner) [inline pep-0670.rst:458]:
```suggestion
This PEP does not propose converting macros which can be used as l-value
```

File: pep-0670.rst
Comment (encukou) [reply pep-0670.rst:85]:
```suggestion
```
“used for definitions rather than behavior” is a better reason to exclude this one, IMO. It does fit both :)

**Comment by vstinner** [comment]
Please wait for Erlend's (@erlend-aasland) approval, since he is one of two authors of this PEP :-) cc @erlend-aasland

File: pep-0670.rst
Comment (vstinner) [reply pep-0670.rst:20]: Yep, but the parenthesis is not needed. It's an important information.

**Review by erlend-aasland — APPROVED** [review]
AFAICS, the major part of this PR is a rearrangement of some of the sections. I'm fine with that. The rewordings are ok to me as well.

Thanks Petr and Victor.

**Pull request merged**
